### ISSUE TITLE
STAC REST API / Browsing / Eodag docker-compose up

### ISSUE THREAD

**Issue by BasileGoussard** [CS-SI/eodag#310]
**Describe the bug**
Failed to build the stac browsing app by using docker-compose with respect to the tutorial 
```
git clone https://github.com/CS-SI/eodag.git
cd eodag
docker-compose up
```
I got two issues.
One concerning the stac-browser.dockerfile which explain that ```node:lts-jessie-slim``` is not found
A second one concerning the installation of  ```shapely``` during the construction of the stac-server.dockerfile

**Code To Reproduce**
CLI commands or Python code snippet to reproduce the bug. Please use maximum verbosity using:
```
git clone https://github.com/CS-SI/eodag.git
cd eodag
docker-compose up
```

**Output**
First bug : 
```
 => ERROR [internal] load metadata for docker.io/library/node:lts-jessie-  1.7s
 => [auth] library/node:pull token for registry-1.docker.io                0.0s
 => [auth] library/nginx:pull token for registry-1.docker.io               0.0s
------
 > [internal] load metadata for docker.io/library/node:lts-jessie-slim:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: no match for platform in manifest sha256:d8dd20f9c00049c6f874683bc6c55cebddcadceca8f82b98975cc9f750bef5db: not found
ERROR: Service 'browser' failed to build
```

Solved by changing ```node:lts-jessie-slim``` to ```node:lts-buster-slim``` 

Second bug : 
```
ERROR [10/14] RUN python setup.py install 
#15 15.21 Best match: Shapely 1.8a1
#15 15.21 Processing Shapely-1.8a1.tar.gz
#15 15.22 Writing /tmp/easy_install-gqproqkm/Shapely-1.8a1/setup.cfg
#15 15.22 Running Shapely-1.8a1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-gqproqkm/Shapely-1.8a1/egg-dist-tmp-xias8_z0
#15 15.31 Failed `CDLL(libgeos_c.so.1)`
#15 15.31 Failed `CDLL(libgeos_c.so)`
#15 15.32 error: Could not find library geos_c or load any of its variants ['libgeos_c.so.1', 'libgeos_c.so']
------
executor failed running [/bin/sh -c python setup.py install]: exit code: 1
ERROR: Service 'stac' failed to build
```
Solved by adding ```RUN apt-get install libgeos-c1v5 libgeos-3.7.1``` before  ```RUN python setup.py install```

**Additional context**
I do not know if it's a relevant way to solved those issues but at least the server runs on my computer.

Ps : 
Thanks for all this work, EODAG is amazing !

**Comment by sbrunato** [comment]
@BasileGoussard many thanks for submitting this issue! We'll use your remarks to fix the `docker-compose` file.
Or if you want, you can also submit a Pull Request , we'll be happy to merge your contribution

**Comment by apparell** [comment]
Hello,

1) it's not a issue about the image source since the lts-jessie-slim is still online: https://hub.docker.com/layers/node/library/node/lts-jessie-slim/images/sha256-72df7b77ceb0935005cbd54b85352538d9c86758acf01eedec40c9b7d45cbda6?context=explore
switching to "lts-buster-slim" seems to resolve some warnings.

2) On the stac-server side, I'm not able to reproduce the errors..

```
Searching for shapely
Reading https://pypi.org/simple/shapely/
Downloading https://files.pythonhosted.org/packages/9d/92/4e9a73043f7e583cb9dbf11e801625fc576dbf37260db7999889435411b5/Shapely-1.8a1-cp39-cp39-manylinux1_x86_64.whl#sha256=80f331b83e102d343ebcae00f615b996d2eaeee698d9b4249cbc1c7b6263014e
Best match: Shapely 1.8a1
Processing Shapely-1.8a1-cp39-cp39-manylinux1_x86_64.whl
Installing Shapely-1.8a1-cp39-cp39-manylinux1_x86_64.whl to /usr/local/lib/python3.9/site-packages
Adding Shapely 1.8a1 to easy-install.pth file

Installed /usr/local/lib/python3.9/site-packages/Shapely-1.8a1-py3.9-linux-x86_64.egg

```

**Comment by sbrunato** [comment]
On my side, same thing as for @apparell : I do not manage to reproduce these issues. 

@BasileGoussard , try rebuilding using  `docker-compose build --no-cache`.

It may be related to your docker environment. Can you try some of the solutions pointed in https://github.com/docker/buildx/issues/476 ? 
- `docker logout & docker login`
- `sudo service docker restart`
-  ...

**Comment by sbrunato** [comment]
Hello @BasileGoussard , 

can you try to rebuild using the latest `develop` branch. I have updated the stac-browser dockerfile (see #314). It uses now `node:lts-buster-slim`.

Thank you

**Comment by sbrunato** [comment]
@BasileGoussard , #314 changes have been published in [v2.3.1](https://github.com/CS-SI/eodag/releases/tag/v2.3.1), please check if this fixes your issue.
Thanks

**Comment by BasileGoussard** [comment]
Hi @sbrunato,

That's fixed the issue :) !  
Sorry for the late answer. 

Have a nice week-end,
Basile
